client/error.go

```diff
@@ -32,12 +40,45 @@ func IsEndpointNotSpecifiedError(err error) bool {
 	return microerror.Cause(err) == endpointNotSpecifiedError
 }
 
-// notAuthorizedError is used when an API request got a 401 response.
-var notAuthorizedError = &microerror.Error{
-	Kind: "notAuthorizedError",
+// NotAuthorizedError is used when an API request got a 401 response.
+var NotAuthorizedError = &microerror.Error{
+	Kind: "NotAuthorizedError",
 }
 
-// IsNotAuthorizedError asserts notAuthorizedError.
+// IsNotAuthorizedError asserts NotAuthorizedError.
 func IsNotAuthorizedError(err error) bool {
-	return microerror.Cause(err) == notAuthorizedError
+	return microerror.Cause(err) == NotAuthorizedError
+}
+
+// HandleErrors handles the errors known to this package.
+// Handling normally means printing a user-readable error message
+// and exiting with code 1. If the given error is not recognized,
+// the function returns without action.
+func HandleErrors(err error) {
+
+	var headline = ""
+	var subtext = ""
+
+	// V2 client error handling
+	if convertedErr, ok := microerror.Cause(err).(*clienterror.APIError); ok {
+		headline = convertedErr.ErrorMessage
+		subtext = convertedErr.ErrorDetails
+	} else if convertedErr, ok := err.(*clienterror.APIError); ok {
+		headline = convertedErr.ErrorMessage
+		subtext = convertedErr.ErrorDetails
+	} else if IsEndpointNotSpecifiedError(err) {
+		// legacy client error handling
+		headline = "No endpoint has been specified."
+		subtext = "Please use the '-e|--endpoint' flag or select an endpoint using 'gsctl select endpoint'."
+	}
+
+	if headline == "" {
+		return
+	}
+
+	fmt.Println(color.RedString(headline))
+	if subtext != "" {
+		fmt.Println(subtext)
+	}
+	os.Exit(1)
```

Comment:
I had to split the big `errors.HandleCommonErrors` due to cyclic imports. Now there is new function for printing common errors occurring in the `client` package.

Reply:
Seems like a sensible change - keeps everything nice and localised to the package